### PR TITLE
docs(CSS): Anchor elements are not referring to '<a>' but to the anchoring element

### LINKED ISSUE #22917: Section on "relative selectors" mentions "anchor element" with incorrect link?
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors

### What specific section or headline is this issue about?

Structure of a selector

### What information was incorrect, unhelpful, or incomplete?

The paragraph under "Relative selectors" says:

> A selector that represents an element relative to one or more anchor elements preceded by a combinator. Relative selectors that don't begin with an explicit combinator have an implied descendant combinator.

The words "anchor elements" is linked to [the page on pseudo-classes](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes). Is this a mistake? What is meant here by "anchor elements"? It doesn't seem to be the anchor element/tag (i.e. `<a>`). I'm not sure what it should be linked to or if the wording needs to be changed.

### What did you expect to see?

Not sure, the link is incorrect and the context is unclear as to what was meant.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/css_selectors`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/css_selectors/index.md
* Last commit: https://github.com/mdn/content/commit/2393cdd9715c0b42a575bf9b8f8c2d95457e5294
* Document last modified: 2022-11-04T23:04:38.000Z

</details>

### PR DESCRIPTION
**Pull request by bsmth**
We had some changes land recently that pointed to anchor elements (`<a>`) where the docs are actually referring to "relative-selector-anchor-elements" -> https://drafts.csswg.org/selectors/#relative-selector-anchor-elements

From the spec:

| CSS | Description|
|---|---|
| `E:has(rs1, rs2)` | an E element, 					if there exists an element that matches 					either of the relative selectors rs1 or rs2, 					when evaluated with E as the anchor elements

Fixes #22917

### CODE REVIEW

File: files/en-us/web/css/css_selectors/index.md [hunk 1]
```diff
@@ -149,9 +149,9 @@ The term 'selector' can refer to one of the following:
 
 - Relative selector
 
-  - : A selector that represents an element relative to one or more [anchor elements](/en-US/docs/Web/HTML/Element/a) preceded by a combinator. Relative selectors that don't begin with an explicit [combinator](#combinators) have an implied [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator).
+  - : A selector that represents an element relative to one or more [anchor elements](/en-US/docs/Web/CSS/Pseudo-classes) (the element that a pseudo-class is attached to) preceded by a combinator. Relative selectors that don't begin with an explicit [combinator](#combinators) have an implied [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator).
```
Comment (estelle) [inline files/en-us/web/css/css_selectors/index.md:152]: Good catch. However, this isn't a good definition. Let me marinate on a good suggestion (it's not coming to me at the moment). Please don't merge as is.

Comment (estelle) [inline files/en-us/web/css/css_selectors/index.md:152]:
```suggestion
  - : A selector representing an element relative to one or more anchor elements preceded by a combinator. Relative selectors that don't begin with an explicit [combinator](#combinators) have an implied [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator).
```

Let's start with this, then add a glossary for` anchor` element perhaps. The /en-US/docs/Web/CSS/Pseudo-classes defines it well, but that's not a good link here (user has to search the entire page to find what they're looking for), and a relative selector doesn't need to include a pseudoclass, so meh.

File: files/en-us/web/css/css_selectors/index.md [hunk 2]
```diff
@@ -149,9 +149,9 @@ The term 'selector' can refer to one of the following:
...
 
-    **Examples:** `+ div#topic > #reference {...}`, `> .icon {...}`
+    **Examples:** `+ div#topic > #reference {...}`, `> .icon {...}`, `dt:has(+ img) {...}`
```
Comment (estelle) [inline files/en-us/web/css/css_selectors/index.md:154]:
```suggestion
    **Examples:** `+ div#topic > #reference {...}`, `> .icon {...}`, `dt:has(+ img) ~ dd {...}`
```

**Review by estelle — CHANGES_REQUESTED** [review]
Good catch. Let's revert the changes. I think it's better to not add a link if it's not a link directly to the definition

File: files/en-us/web/css/css_selectors/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/css/css_selectors/index.md]: Good idea

> add a glossary for anchor element perhaps

Should we do this in this PR or a follow-up?

**Comment by bsmth** [comment]
Thanks @estelle, incorporated your suggestions 👍🏻

**Review by estelle — APPROVED** [review]

**Pull request merged**